torch/nested/_internal/ops.py

```diff
@@ -161,15 +192,17 @@ def jagged_binary_pointwise(func, *args, **kwargs):
     # === Handle broadcasting across the batch / ragged dims ===
 
     # Easy case: take advantage of pre-existing broadcasting logic
-    # when NT dim > non-NT dim
-    # ex: (B, j0, D_0, D_1) + (D_0, D_1) -> (B, j0, D_0, D_1)
-    # ex: (B, j0, D_0, D_1) + (1, D_0, D_1) -> (B, j0, D_0, D_1)
-    # ex: (B, j0, 1, 1) + (D_0, D_1) -> (B, j0, D_0, D_1)
-    # ex: (B, j0, 1, 1) + (1, D_0, D_1) -> (B, j0, D_0, D_1)
-    if (a_is_nt and a.dim() > b.dim()) or (not a_is_nt and b.dim() > a.dim()):
-        arg1 = a._values if a_is_nt else a
-        arg2 = b._values if not a_is_nt else b
-        return NestedTensor(func(arg1, arg2, *args[2:], **kwargs), **extracted_kwargs)
+    # ex: (B, j0, ?, ?) + (?) -> (B, j0, ?, ?)
+    # ex: (B, j0, ?, ?) + (?, ?) -> (B, j0, ?, ?)
+    # ex: (B, j0, ?, ?) + (1, 1, ?, ?) -> (B, j0, ?, ?)
+    nt, t = (a, b) if a_is_nt else (b, a)
+    # See Note: [ Squeezing leading ones ]
+    t_squeezed, extra = squeeze_leading_ones_get_extra(t, nt)
```

Comment:
dumb Q: can you explain a bit more the purpose behind this change? I'm unclear as to what it's solving in the context of this PR

Reply:
Previously we were too relaxed when it comes to checking whether the pre-existing broadcasting logic is valid for the given NTs. e.g. someone could try `JT: [B, *, D] + T: [sum(*), D]` and still produce something, even though broadcasting didn't really make sense in the first place. 

What we kind of what to do is to make sure that the JT has a dim that is two greater than that of the other tensor, so that we would have something like `(B, j0, a0, ..., an) + (b0, ..., bm) where m < n` where B and j0 are always broadcasted over uniformly (as in any value in a given batch acquires the same values) during broadcasting logic. This is important because otherwise we would fall into the unbind case below.

We want this `squeeze_leading_ones_get_extra` helper because naively doing `nt.dim() > t.dim() + 2` doesn't quite work though because the t might have leading ones in its dim that shouldn't disqualify it from the easy case.

Reply:
cool, thanks for the explanation :)

Reply:
Note that if we see `t.dim() > nt.dim()`, we know we're in an unsupported case, even if `t` is all ones. If we bail out early for this, it might be a little more readable.